geoengine/datasets.py

```diff
@@ -266,8 +266,8 @@ def from_response(cls, response: api.DatasetId) -> DatasetId:
         if 'error' in response:
             raise GeoEngineException(cast(api.GeoEngineExceptionResponse, response))
 
-        if 'id' not in response:  # TODO: improve error handling
-            raise Exception('No id in response')
+        if 'id' not in response:
+            raise UnexpectedResponseException('No id in response', response)
```

Comment:
maybe a MissingFieldInResponseExcpetion would be better?

Reply:
[d825fa0](https://github.com/geo-engine/geoengine-python/pull/102/commits/d825fa04cb26a20bebd9fbab2e1aa3d54d18af97)

Reply:
I had a typo in there "Excpetion" which you copied. Sorry about that.

Reply:
[26fad8d](https://github.com/geo-engine/geoengine-python/pull/102/commits/26fad8dbd3c6f7319694f93d089d6cc43800e802)